addons/account/models/account_journal.py

```diff
@@ -408,35 +466,69 @@ def _check_payment_method_line_ids_multiplicity(self):
-
-        if not unique_codes:
-            return
-
         self.flush(['inbound_payment_method_line_ids', 'outbound_payment_method_line_ids', 'company_id'])
-        self.env['account.payment.method.line'].flush(['payment_method_id', 'journal_id'])
+        self.env['account.payment.method.line'].flush(['payment_method_id', 'journal_id', 'name'])
         self.env['account.payment.method'].flush(['code'])
 
-        if unique_codes:
-            self._cr.execute('''
-                SELECT apm.id
-                FROM account_payment_method apm
-                JOIN account_payment_method_line apml on apm.id = apml.payment_method_id
-                JOIN account_journal journal on journal.id = apml.journal_id
-                JOIN res_company company on journal.company_id = company.id
-                WHERE apm.code in %s
-                GROUP BY
-                    company.id,
-                    apm.id
-                HAVING array_length(array_agg(journal.id), 1) > 1;
-            ''', [unique_codes])
-
-        method_ids = [res[0] for res in self._cr.fetchall()]
-        if method_ids:
-            methods = self.env['account.payment.method'].browse(method_ids)
-            raise ValidationError(_("Some payment methods supposed to be unique already exists somewhere else.\n"
-                                    "(%s)", ', '.join([method.display_name for method in methods])))
+        results = self._get_journals_payment_method_information()
+        pay_methods = results['pay_methods']
+        manage_acquirers = results['pay_methods']
+        method_information_mapping = results['method_information_mapping']
+        acquirers_per_code = results['acquirers_per_code']
+
+        failing_unicity_payment_methods = self.env['account.payment.method']
+        for journal in self:
+            company = journal.company_id
+
+            # Exclude the 'unique' / 'electronic' values that are already set on the journal.
+            protected_acquirer_ids = set()
+            protected_payment_method_ids = set()
+            for payment_type in ('inbound', 'outbound'):
+                lines = journal[f'{payment_type}_payment_method_line_ids']
+
+                # Ensure you don't have twice the same combination payment_method/name on the same journal.
+                counter = {}
+                for line in lines:
+                    if method_information_mapping[line.payment_method_id.id]['mode'] not in ('electronic', 'unique'):
+                        continue
+
+                    key = line.payment_method_id.id, line.name
+                    counter.setdefault(key, 0)
+                    counter[key] += 1
+                    if counter[key] > 1:
+                        raise ValidationError(_(
+                            "You can't have two payment method lines of the same payment type (%s) "
+                            "and with the same name (%s) on a single journal.",
```

Comment:
```suggestion
                            "using the same name (%s) on a single journal.",
```

Reply:
@epictete I kept the old term to not break the translations